webapp/src/lib/Ui.ts

```diff
@@ -24,42 +32,79 @@ export async function showSnack(snack: MatSnackBar, text: string, cmd: () => voi
+ * method is a boolean: true for OK and false for Cancel.
+ *
+ * @param dialog reference to the matDialog
+ * @param title shown in h1 in the dialog
+ * @param text the text in HTML
+ * @param buttons can override the text of the OKButton ("OK") or the CancelButton ("Cancel")
+ */
 export async function showDialogOKC(dialog: MatDialog, title: string, text: string,
-                                    result: (result: boolean) => void,
-                                    buttons: IDialogOKCButtons = {OKButton: "OK", CancelButton: "Cancel"}) {
+                                    buttons: IDialogOKCButtons = {OKButton: "OK", CancelButton: "Cancel"}):
+    Promise<boolean> {
     const tc = dialog.open(DialogOKCancelComponent, {data: {Title: title, Text: text, Buttons: buttons}});
-    tc.afterClosed().subscribe(result);
+    return new Promise((resolve) => tc.afterClosed().subscribe((result) => resolve(result)));
 }
 
+/**
+ * Shows a simple dialog with a dismiss button and returns when the user presse the dismiss button.
+ *
+ * @param dialog reference to the matDialog
+ * @param title shown in h1 in the dialog
+ * @param text the text in HTML
+ * @param dismiss the string to be shown in the dismiss button
+ */
 export async function showDialogInfo(dialog: MatDialog, title: string, text: string,
-                                     dismiss: string,
-                                     result: (result: boolean) => void = null) {
+                                     dismiss: string) {
     const tc = dialog.open(DialogOKCancelComponent, {
         data: {
             Buttons: {OKButton: dismiss, CancelButton: ""},
             Text: text,
             Title: title,
         },
     });
-    tc.afterClosed().subscribe((res) => {
-        if (result) {
-            result(res);
-        }
-    });
+    return new Promise((resolve) => tc.afterClosed().subscribe(() => resolve()));
 }
 
+/**
+ * Convenience method for showTransactions that only shows one progress step: "Storing Credential".
+ *
+ * @param dialog reference to the matDialog
+ * @param title shown in h1 in the dialog
+ * @param store the callback to the actual storing of the credential.
+ */
 export async function storeCredential(dialog: MatDialog, title: string,
-                                      store: () => Promise<void>) {
+                                      store: <T>() => Promise<T | void>) {
     return showTransactions(dialog, title, async (progress: TProgress) => {
         progress(50, "Storing Credential");
-        await store();
+        return store();
     });
 }
 
+// Progress type to be used in showTransactions.
 export type TProgress = (percentage: number, text: string) => void;
-export type TWorker = (progress: TProgress) => Promise<void>;
+// Worker callback that implements multiple steps and calls progress before each step.
+export type TWorker<T> = (progress: TProgress) => Promise<T | void>;
```

Comment:
```suggestion
export type TWorker<T> = (progress: TProgress) => Promise<T>;
```

Reply:
Most of the calls to `showTransaction` don't return anything, this is why I would like to keep the `void`...

Reply:
Yup, and `T` already ~contains `void`, as it represent any type (there is no type bounds). The compiler reduce `T | void` to `T` anyway.

Reply:
yes, but then I need to

```
return void/null/undefined
```

at the end of my callbacks :(

Reply:
You're always right, dear...